framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -1025,10 +1026,14 @@ public void postProcessClassTree(ClassTree tree) {
         TypesIntoElements.store(processingEnv, this, tree);
         DeclarationsIntoElements.store(processingEnv, this, tree);
         if (checker.hasOption("infer") && wholeProgramInference != null) {
-            // Write scenes into .jaif files. In order to perform the write
-            // operation only once for each .jaif file, the best location to
-            // do so is here.
-            wholeProgramInference.saveResults();
+            // Write out the results of whole-program inference.
+            // In order to perform the write operation only once for each class, the best location
+            // to do so is here.
+            WholeProgramInference.OutputKind outputKind =
```

Comment:
Maybe this should be extracted to where field `wholeProgramInference` is set, to avoid re-computing this on every classtree.